core/src/test/java/kafka/testkit/KafkaClusterTestKit.java

```diff
@@ -638,6 +634,7 @@ public void close() throws Exception {
             throw e;
         } finally {
             ThreadUtils.shutdownExecutorServiceQuietly(executorService, 5, TimeUnit.MINUTES);
+            waitForAllThreads();
```

Comment:
we should not throw exception in the final block, since it will abort the current exception